src/collects/seashell/backend/runner.rkt

```diff
@@ -111,8 +116,18 @@
        ;; Read stdout, stderr.
        (close-output-port cp-stderr)
        (close-output-port cp-stdout)
+       ;; Read asan error message.
+       ;; For tests (instead of clicking run button), we append a special string
+       ;; so that the front end (console.js) can extract it and parse the json string.
+       ;; Otherwise, students will see a bunch of jibberish.
+       (define asan-marker-string #"this_is_a_special_asan_marker_string_7f84028cdd719df7570532dbf54cd10c3b5a1cc1b4bb62193079da701fdd9acb")
```

Comment:
You can probably pass this out as another entry in the serialize call on Line 137, and read it back in using the match on Line 102 of dispatch.rkt.